.github/workflows/tests.yaml

```diff
@@ -0,0 +1,61 @@
+name: Tests
+
+on:
+    push:
+        branches: [ main, master ]
+    pull_request:
+    release:
+        types: [ created ]
+
+jobs:
+    unit-tests:
+        runs-on: ${{ matrix.operating-system }}
+        strategy:
+            matrix:
+                operating-system: [ ubuntu-latest ]
+                php-versions:
+                    - '7.2'
+                    - '7.3'
+                    - '7.4'
+                    - '8.0'
+                    - '8.1'
+        name: PHP ${{ matrix.php-versions }} Test on ${{ matrix.operating-system }}
+
+        steps:
+            -   name: Checkout
+                uses: actions/checkout@v3
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: ${{ matrix.php-versions }}
+                    extensions: mbstring, intl, zip, xml
+                    coverage: none
+
+            -   name: Install dependencies
+                uses: ramsey/composer-install@v2
+
+            -   name: Update PHPUnit
+                run: composer update phpunit/phpunit --with-dependencies
+
+            -   name: Run test suite
+                run: vendor/bin/phpunit --exclude-group mayignore --coverage-clover build/logs/clover.xml
+
+
+    # This is a "trick", a meta task which does not change, and we can use in
```

Comment:
I don't ensure the meta task is necessary. And I think it should print some status messages before running the `exit` command.

Reply:
if you are curious, this looks like [this](https://github.com/theofidry/console/actions/runs/6292164585).

You can see that "Test 8.1; Symfony 5.4" which change based on the matrix values (of course they could be fixed but then you loose information).

If you want to add a protection rule to your branch that all the tests must pass, you will need to list them all. So you're rule will look like this:

<img width="811" alt="Screenshot 2023-09-30 at 22 19 48" src="https://github.com/phpbrew/phpbrew/assets/5175937/99a24c7f-e272-4d60-bdc9-b26d1a500750">

This is not ideal because it means whenever you change the matrix parameters, your rule configuration becomes outdated. Instead the above allows you to do something like this:

<img width="811" alt="Screenshot 2023-09-30 at 22 20 43" src="https://github.com/phpbrew/phpbrew/assets/5175937/7f499e26-73ab-470a-94a7-5c4cc6321e2c">

Then it doesn't matter what value you have in your matrix, as long as they all pass, the check status will pass. If any fail, it won't, and the rule will always be up to date.

Reply:
@peter279k not convinced? Should we close this then?

Reply:
Go ahead with this and I think there's no problem now.